### ISSUE TITLE
Slowing typing performance

### ISSUE THREAD

**Issue by xoyojank** [tgjones/HlslTools#183]
![image](https://user-images.githubusercontent.com/2873643/71244038-9acf8780-234c-11ea-85dd-a73323f2cb8c.png)
VS2017 Professional, Edit UnrealEngine usf files

**Comment by tgjones** [comment]
Please could you try disabling all auto-formatting for HLSL - as in the attached screenshot - and see if it fixes the performance problem?

![image](https://user-images.githubusercontent.com/102049/71309887-1260ed80-2405-11ea-887e-882ad3b585dd.png)

**Comment by jamiehayes** [comment]
Hi, Tim! I'm also getting rough typing performance from this plugin in VS 2022 in Unreal Engine 5 projects. Often just adding a newline to a shader will cause VS to pop up this box, followed by the warning toast at the top that the HLSL Tools plugin is causing slow typing performance.

![image](https://user-images.githubusercontent.com/4496010/170776249-b9717fc8-3675-4250-91fe-3350db1e943f.png)

I have disabled formatting when typing/paste, Intellisense, and error reporting, but the issue doesn't go away. It's a shame, because I do really like the plugin, but the typing perf is going to be the reason I have to uninstall it.

Let me know if there's any other data I can provide to help you reproduce and fix the issue! Would love to get a new drop that fixes this.

-Jamie

**Comment by DFMattE** [comment]
Agreed, default shader editing in Unreal Engine 5 usf/ush files is slow and can make typing difficult. For example, holding space down has a noticeable delay between auto-repeating spaces.

Turning off "Auto list members" in the HLSL > General settings improves the typing speed significantly, though there are still occasional delays that only go away when the plugin is disabled.

**Comment by Wunkolo** [comment]
Going to have to add onto this too. Almost every interaction I have with an unreal engine shader-file is extremely slow or just causes the `editor command to finish...` message to show up and lock me out of visual studio. Sometimes it just freezes visual studio without the `editor command to finish...` message and I have to force-close `devenv.exe` from task manager, destroying my debug-context in some cases.
